transformer_engine/common/include/transformer_engine/gemm.h

```diff
@@ -51,6 +51,10 @@ void nvte_cublas_gemm(const NVTETensor A,
                       bool accumulate,
                       bool use_split_accumulator,
                       int math_sm_count,
+                      int m_split,
+                      int n_split,
+                      bool gemm_producer,
+                      const NVTETensor counter,
```

Comment:
I think i would prefer this to be its own API rather than adding new parameters to the nvte_cublas_gemm. Maybe nvte_cublas_parallel_gemm or something like that?